### ISSUE TITLE
proxmox_virtual_environment_network_linux_... disable direct appliement

### ISSUE THREAD

**Issue by NiclasPe** [bpg/terraform-provider-proxmox#1637]
**Describe the bug**
I wanna create all VLANS and Bridges via Terraform to keep it consistent over multiple Nodes, but if there are a few of this resources, Terraform will fail due to the netplan apply for each resource. It would be great if there is a option to disable the direct apply in the resource and apply all changes at the end of the script.

**To Reproduce**

1. Create many resources 'proxmox_virtual_environment_network_linux_bridge' maybe with vlan dependencies
 -> Didn't  happens every time, but if you use 10 of them over 2 nodes most of the time
2. Run 'terraform apply'
3. See error

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl

erraform {
  required_providers {
    proxmox = {
      source  = "bpg/proxmox"
      version = "0.66.2"
    }
  }
}


provider "proxmox" {
  endpoint  = "https://10.10.2.20:8006/"
  api_token = "root@pam!terraform=abc"
  insecure  = true
  ssh {
    agent    = true
    username = "root"
    node {
      name    = "pve1"
      address = "10.10.2.23"
    }
    node {
      name    = "pve2"
      address = "10.10.2.21"
    }
    node {
      name    = "pve3"
      address = "10.10.2.22"
    }
  }
}


#### Locals ####

locals {
  nodes = toset([
    "pve1",
    "pve2",
    "pve3"
  ])
}

#### Resources ####

# VLAN
resource "proxmox_virtual_environment_network_linux_vlan" "bond0_10" {
  for_each  = local.nodes
  node_name = each.value

  name    = "bond0.10"
  comment = "VLAN - Guests"
}

# Bridge
resource "proxmox_virtual_environment_network_linux_bridge" "vmbr10" {
  depends_on = [
    proxmox_virtual_environment_network_linux_vlan.bond0_10
  ]

  for_each  = local.nodes
  node_name = each.value

  name    = "vmbr10"
  comment = "Network - Guests"
  ports = [
    "bond0.10"
  ]
}

# ---------------------------------------------------------------------------------------------

# VLAN
resource "proxmox_virtual_environment_network_linux_vlan" "bond0_11" {
  for_each  = local.nodes
  node_name = each.value

  name    = "bond0.11"
  comment = "VLAN - LAN"
}

# Bridge
resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
  depends_on = [
    proxmox_virtual_environment_network_linux_vlan.bond0_11
  ]

  for_each  = local.nodes
  node_name = each.value

  name    = "vmbr0"
  comment = "Network - LAN"
  ports = [
    "bond0.11"
  ]
}

...

```

**Expected behavior**
Runs without timeout failures

**Additional context**
Happens because of the direct apply for each resource. If there is a possibility to disable it, it would run smooth. 

- Clustered Proxmox:
- Proxmox version: latest
- Provider version (ideally it should be the latest version): 0.66.2
- Terraform/OpenTofu version: v1.5.7
- OS (where you run Terraform/OpenTofu from): MacOS or Linux

**Comment by bpg** [comment]
Hi @NiclasPe 👋 
What's an error do you see?

**Comment by bpg** [comment]
Hey @NiclasPe 👋🏼 

Terraform can't "postpone" applying the configuration in that way, as each resource is treated independently and is applied as a whole or not at all.

I think the issue you're facing is the parallelism of network changes during execution. I see you have multiple interfaces defined for each of your nodes as separate resources, e.g., `bond0_4001`, `bond0_4002`, `bond0_4003`, `bond0_4004`. Since there are no dependencies between them, Terraform applies them in parallel, using the current `parallelism` setting, which I believe is `4` by default. This means four concurrent network changes are executed at roughly the same time on the same node, and each triggers a network restart at the end of the apply, increasing the likelihood of breaking things.

You have two options to mitigate this:

1. Create a dependency chain for all `vmbr*` and `bond0_*` resources. For example, establish dependencies like `vmbr0 -> vmbr1 -> ...` and `bond0_1 -> bond0_2 -> ...`. The specific order within the chain doesn't matter—it simply ensures Terraform processes these resources sequentially rather than in parallel. You can retain the existing `vmbrX -> bondX` dependency to ensure all "bond" resources are created first.

2. Set [`parallelism`](https://developer.hashicorp.com/terraform/cli/commands/apply#parallelism-n) to `1` when running `apply`.

Please let me know if that solves your issue.

**Comment by bpg** [comment]
No feedback from the author, closing the ticket.

**Comment by Budy-Love** [comment]
Sry for bringing up this on the table again. I'm facing the same problem. The parallelism works but feels a bit strange to me. I would prefer the dependency chain. Can u a bit more explain how to achieve this?

**Comment by bpg** [comment]
@Budy-Love could you share your config?

**Comment by Budy-Love** [comment]
Sure, thats config:

```
resource "proxmox_virtual_environment_network_linux_vlan" "vlan" {
  for_each      = { for entry in local.vlan_nodes: "${entry.node}.${entry.vlan}" => entry }
  node_name = each.value.node
  name      = "vlan${each.value.vlan}"
  interface = "bond0"
}

resource "proxmox_virtual_environment_network_linux_bridge" "vmbr" {
  depends_on = [
    proxmox_virtual_environment_network_linux_vlan.vlan
  ]
  for_each      = { for entry in local.vlan_nodes: "${entry.node}.${entry.vlan}" => entry }
  node_name = each.value.node
  name      = "vmbr${each.value.vlan}"
  ports = [
    "vlan${each.value.vlan}"
  ]
}
```
```
locals {
  vlans = [
    "100",
    "102",
    "300",
    "500",
    "900",
    "1100",
    "1110",
    "1200",
    "1210",
    "1300",
    "1310",
    "1800",
    "1900",
  ]
  nodes = [
    "pve-1",
    "pve-2",
    "pve-3",
  ]
# # # vlan_nodes = tolist([
# # #   {
# # #     "node" = "pve-1"
# # #     "vlan" = "100"
# # #   },
# # #   {
# # #     "node" = "pve-1"
# # #     "vlan" = "102"
# # #   },

  vlan_nodes = distinct(flatten([
    for node in local.nodes : [
      for vlan in local.vlans : {
        node = node
        vlan    = vlan
      }
    ]
  ]))
}
```

**Comment by Budy-Love** [comment]
Thanks for clarification!
